pkg/util/kvgraph/util.go

```diff
@@ -73,20 +73,40 @@ func addVolumeGraph(vmiSpec v1.VirtualMachineInstanceSpec, vmName, namespace str
 		case volume.DataVolume != nil:
 			resources = addVeleroResource(volume.DataVolume.Name, namespace, "datavolumes", resources)
 			resources = addVeleroResource(volume.DataVolume.Name, namespace, "persistentvolumeclaims", resources)
+
 		case volume.PersistentVolumeClaim != nil:
 			resources = addVeleroResource(volume.PersistentVolumeClaim.ClaimName, namespace, "persistentvolumeclaims", resources)
+
 		case volume.MemoryDump != nil:
 			resources = addVeleroResource(volume.MemoryDump.ClaimName, namespace, "persistentvolumeclaims", resources)
+
 		case volume.ConfigMap != nil:
 			resources = addVeleroResource(volume.ConfigMap.Name, namespace, "configmaps", resources)
+
 		case volume.Secret != nil:
 			resources = addVeleroResource(volume.Secret.SecretName, namespace, "secrets", resources)
+
 		case volume.ServiceAccount != nil:
 			resources = addVeleroResource(volume.ServiceAccount.ServiceAccountName, namespace, "serviceaccounts", resources)
+
+		case volume.CloudInitNoCloud != nil:
+			if volume.CloudInitNoCloud.UserDataSecretRef != nil {
+				resources = addVeleroResource(volume.CloudInitNoCloud.UserDataSecretRef.Name, namespace, "secrets", resources)
+			}
+			if volume.CloudInitNoCloud.NetworkDataSecretRef != nil {
+				resources = addVeleroResource(volume.CloudInitNoCloud.NetworkDataSecretRef.Name, namespace, "secrets", resources)
+			}
+
+		case volume.CloudInitConfigDrive != nil:
+			if volume.CloudInitConfigDrive.UserDataSecretRef != nil {
+				resources = addVeleroResource(volume.CloudInitConfigDrive.UserDataSecretRef.Name, namespace, "secrets", resources)
+			}
+			if volume.CloudInitConfigDrive.NetworkDataSecretRef != nil {
+				resources = addVeleroResource(volume.CloudInitConfigDrive.NetworkDataSecretRef.Name, namespace, "secrets", resources)
+			}
 		}
 	}
-	// Returning full backup even if there was an error retrieving the backend PVC.
```

Comment:
Please don't remove this comment unless there's some reason related to the PR. This behavior still aplies.

Reply:
Thanks.

- Squashed commits into one.
- Retained the comment that was unintentionally removed.